examples/language_model/gpt-3/static/modeling.py

```diff
@@ -830,3 +850,175 @@ def forward(self, prediction_scores, masked_lm_labels, loss_mask):
         masked_lm_loss = paddle.sum(masked_lm_loss.reshape([-1]) * loss_mask)
         loss = masked_lm_loss / loss_mask.sum()
         return loss
+
+
+class GPTForGeneration(GPTPretrainedModel):
+    def __init__(self,
+                 gpt,
+                 max_length=20,
+                 min_length=0,
+                 decoding_strategy='sampling',
+                 temperature=1.0,
+                 top_k=0,
+                 top_p=1.0,
+                 eos_id=None):
+        super(GPTForGeneration, self).__init__()
+        self.gpt = gpt
+        self.apply(self.init_weights)
+        self.vocab_size = gpt.vocab_size
+        self.eos_token_id = eos_id or 7
+
+        self.min_dec_len = min_length
+        self.max_dec_len = max_length
+        self.decoding_strategy = decoding_strategy
+        self.temperature = temperature
+        self.topk = top_k
+        self.topp = top_p
+
+    def parallel_matmul(self, lm_output, logit_weights, parallel_output, topo):
+        if topo is not None and topo.mp_info.size > 1:
+            input_parallel = paddle.distributed.collective._c_identity(
+                lm_output, group=None)
+
+            logits = paddle.matmul(
+                input_parallel, logit_weights, transpose_y=True)
+
+            if parallel_output:
+                return logits
+
+            paddle.distributed.init_parallel_env()
```

Comment:
```suggestion
            paddle.distributed.init_parallel_env()
```
init_parallel_env 放这里有额外的考虑吗，可否在模型启动时？
会不会出现调用多次 init_parallel_env？

Reply:
去掉了，debug时加的，当前collective._c_concat不支持静态图，后续还需在框架内部修复bug。